### PR TITLE
Change error type produced by duration parser

### PR DESCRIPTION
**Pull request by iosmanthus**
## What have you changed? (mandatory)
Change the error type produced by `Duration` which make the conversion of `Duration` easier.

## What are the type of changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- Improvement (a change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
No need to test. `cargo check --all-targets`

### CODE REVIEW

File: components/tidb_query/src/codec/mysql/duration.rs [hunk 1]
```diff
@@ -10,6 +10,7 @@ use tikv_util::codec::BytesSlice;
 
 use super::{check_fsp, Decimal};
 use crate::codec::convert::ConvertTo;
+use crate::codec::error::Error;
 use crate::codec::mysql::MAX_FSP;
 use crate::codec::{Result, TEN_POW};
```
Comment (breeswish) [inline components/tidb_query/src/codec/mysql/duration.rs]: codec::{..., Error, ...}

**Comment by breeswish** [comment]
CI failed

File: components/tidb_query/src/codec/mysql/duration.rs [hunk 2]
```diff
@@ -29,14 +29,14 @@ const MAX_MINUTES: u32 = 59;
...
+
 #[inline]
 fn check_hour(hour: u32) -> Result<u32> {
     if hour > MAX_HOURS {
-        Err(invalid_type!(
-            "invalid hour value: {} larger than {}",
-            hour,
-            MAX_HOURS
-        ))
+        Err(Error::overflow(hour, MAX_HOURS))
```
Comment (sticnarf) [inline components/tidb_query/src/codec/mysql/duration.rs]: I find the `Error::overflow` code is:
```
pub fn overflow(data: impl Display, expr: impl Display) -> Error {
    let msg = format!("{} value is out of range in '{}'", data, expr);
    Error::Eval(msg, ERR_DATA_OUT_OF_RANGE)
}
```

It doesn't seem correct to me that `MAX_HOURS` is what the `expr` is expected to be here.

File: components/tidb_query/src/codec/mysql/duration.rs [hunk 3]
```diff
@@ -449,7 +437,7 @@ impl Duration {
         Duration::from_micros(
             millis
                 .checked_mul(1000)
-                .ok_or(invalid_type!("micros overflow"))?,
+                .ok_or_else(|| Error::overflow(millis, MAX_TIME_IN_SECS * 1000))?,
```
Comment (sticnarf) [inline components/tidb_query/src/codec/mysql/duration.rs]: ditto

**Pull request merged**